src/utils/helpers/structure/generateReviewValidity.ts

```diff
@@ -0,0 +1,41 @@
+import { Decision } from '../../generated/graphql'
+import { FullStructure } from '../../types'
+import { getChangeRequestsPropress } from './generateReviewerChangesRequestedProgress'
+import { getReviewProgress } from './generateReviewerResponsesProgress'
+
+const generateReviewValidity = (newStructure: FullStructure) => {
+  const sortedPages = newStructure?.sortedPages || []
+  const { doneConform, doneNonConform, totalReviewable } = getReviewProgress(
+    Object.values(newStructure.sections)
+  )
+  const { doneChangeRequests, totalChangeRequests } = getChangeRequestsPropress(
+    Object.values(newStructure.sections)
+  )
+
+  let firstIncompleteReviewPage
+
+  const incompletedUpdatesToChangesRequest = totalChangeRequests > doneChangeRequests
+  const incompletedReviews = doneNonConform === 0 && totalReviewable > doneConform
+
+  if (incompletedUpdatesToChangesRequest || incompletedReviews) {
+    const firstIncomplete = sortedPages.find(
+      ({ reviewProgress }) =>
+        reviewProgress?.totalReviewable !==
+        (reviewProgress?.doneConform || 0) + (reviewProgress?.doneNonConform || 0)
+    )
```

Comment:
I think i had it slightly wrong previously, it should have been just checking for doneConform === totalReviewable.

Now I think it needs to check (for a valid page):

* all doneConform ?
* at least one doneNonConform AND totalChangeRequest === doneChangeRequests ?

I think it needs to be it's own methods with comments, as per below

```suggestion
  const pageIsNotValid = ({ reviewProgress, changeRequestsProgress }: Page) => {
    const { doneChangeRequests = 0, totalChangeRequests = 0 } = changeRequestsProgress || {}
    // Not valid if remaining change requests
    if (doneChangeRequests !== totalChangeRequests) return true
    const { totalReviewable, doneConform } = reviewProgress || {}
    // Valid if all conform
    if (totalReviewable === doneConform) return false
    // Valid if at least one non conform
    if (doneNonConform > 0) return false
    // Otherwise invalid
    return true
  }

  if (incompletedUpdatesToChangesRequest || incompletedReviews) {
    const firstIncomplete = sortedPages.find(pageIsNotValid)

    if (!firstIncomplete) return
    else
      firstIncompleteReviewPage = {
        sectionCode: firstIncomplete.sectionCode,
        pageNumber: firstIncomplete.number,
      }
  }

```